### PR TITLE
Add libmono to Build System

### PR DESCRIPTION
**Pull request by cwfitzgerald**
This PR simply adds CI and CMake machinery for the use of mono in nova. There are probably more issues that need to be solved, but those will come after we start using it.

### CODE REVIEW

File: external/external.cmake
Comment (Janrupf) [inline external/external.cmake]: Those debug message should probably be removed

**Review by Janrupf — CHANGES_REQUESTED** [review]

File: external/external.cmake
Comment (cwfitzgerald) [reply external/external.cmake]: Yup, that escaped me.

**Review by Janrupf — APPROVED** [review]

**Comment by cwfitzgerald** [comment]
Do not merge until deth reviews

File: .github/workflows/ci.yml
Comment (DethRaid) [inline .github/workflows/ci.yml:105]: Yasssss

Comment (cwfitzgerald) [reply .github/workflows/ci.yml:105]: 🙄 😛

File: README.md
Comment (DethRaid) [inline README.md]: I don't really know if mono should be a system dependency. I'm not sure how hard it would be to build ourselves so this is fine for now

File: external/external.cmake
Comment (DethRaid) [inline external/external.cmake]: You are responsible for maintaining this part of the CMake file

File: README.md
Comment (Janrupf) [reply README.md]: We probably don't want to build it on our own, this would take a very long time...

Comment (cwfitzgerald) [reply README.md]: It's fine on linux, but on windows it's a compound msvc (for c++) + cygwin (for c#) build which sounds extraordinarily annoying.

**Review by DethRaid — CHANGES_REQUESTED** [review]
The mono CMake stuff should be in its own file. once that is done we can merge this

File: external/external.cmake
Comment (cwfitzgerald) [reply external/external.cmake]: lmao alright

File: README.md
Comment (DethRaid) [reply README.md]: Alright. In that case let's just say the specific version of mono that we're using

**Review by DethRaid — APPROVED** [review]

File: README.md
Comment (cwfitzgerald) [reply README.md]: Done.

**Comment by cwfitzgerald** [comment]
Closing due to security concerns.

**Pull request closed without merging**
